features/step_definitions/web_steps.rb

```diff
@@ -319,3 +319,9 @@ def with_scope(locator)
   cookie_value = Capybara.current_session.driver.request.cookies.[](cookie)
   assert cookie_value.nil?
 end
+
+When "I fix the domain name" do
+  parsed = URI.parse(current_url)
+  dest = "#{parsed.path}?#{parsed.query || ''}##{parsed.fragment || ''}"
+  visit dest
+end
```

Comment:
I'm not sure exactly what happens, but every time there's a non Capybara page change with JS enabled (clicking a link, getting redirected...), the domain is replaced with `example.com`.

No idea how to get around that issue cleanly honestly.